app/Helpers/database/ticket.php

```diff
@@ -516,22 +509,20 @@ function getNumberOfTicketsClosedForOthers(string $user): array
 /**
  * Gets the number of tickets closed/resolved for achievements written by the user.
  */
-function getNumberOfTicketsClosed(string $user): array
+function getNumberOfTicketsClosed(User $user): array
 {
-    sanitize_sql_inputs($user);
-
     $retVal = [];
     $query = "SELECT ua2.User AS ResolvedByUser, COUNT(ua2.User) AS TicketCount,
               SUM(CASE WHEN t.ReportState = " . TicketState::Closed . " THEN 1 ELSE 0 END) AS ClosedCount,
               SUM(CASE WHEN t.ReportState = " . TicketState::Resolved . " THEN 1 ELSE 0 END) AS ResolvedCount
               FROM Ticket AS t
               LEFT JOIN UserAccounts as ua ON ua.ID = t.reporter_id
               LEFT JOIN UserAccounts as ua2 ON ua2.ID = t.resolver_id
-              LEFT JOIN Achievements as a ON a.ID = t.AchievementID
+              LEFT JOIN Achievements as ach ON ach.ID = t.AchievementID
               WHERE t.ReportState IN (" . TicketState::Closed . "," . TicketState::Resolved . ")
-              AND ua.User NOT LIKE '$user'
-              AND a.Author LIKE '$user'
-              AND a.Flags = '3'
+              AND ua.ID != {$user->id}
+              AND ach.user_id = {$user->id}
+              AND ach.Flags = '3'
```

Comment:
You updated the other lines to use the enum. Recommend doing it here too.

Reply:
Done in latest